pkg/kubelet/cm/dra/plugin/registration.go

```diff
@@ -72,17 +74,16 @@ type RegistrationHandler struct {
 
 var _ cache.PluginHandler = &RegistrationHandler{}
 
-// NewPluginHandler returns new registration handler.
+// NewRegistrationHandler creates a new registration handler.
 //
-// Must only be called once per process because it manages global state.
 // If a kubeClient is provided, then it synchronizes ResourceSlices
 // with the resource information provided by plugins.
-func NewRegistrationHandler(kubeClient kubernetes.Interface, getNode func() (*v1.Node, error), wipingDelay time.Duration) *RegistrationHandler {
+func NewRegistrationHandler(draPlugins *Store, kubeClient kubernetes.Interface, getNode func() (*v1.Node, error), wipingDelay time.Duration) *RegistrationHandler {
```

Comment:
it feels awkward to have this take an already initialized (but empty) `*Store` object which it then caches locally and populates. Does it make sense to have this return `(*RegistrationHandler, *Store)` so that the component that actually populates this object owns this object and the one that consumes it is the one that caches a reference to it.

Reply:
The way I see it, RegistrationHandler is one layer above the central store. It would be possible to use a store without a registration handler, or with some other way of registering endpoints. With that in mind, constructing the store separately seems consistent.

If this feels like overkill, then let's eliminate the RegistrationHandler. The store can implement `cache.PluginHandler` directly.

Reply:
I do not particularly care about RegistrationHandler being separate. I kept it because it was that way before and only clarified the layering.

Reply:
I like the idea of the store implementing cache.PluginHandler directly

Reply:
Well at this point, I'd just argue that maybe `store` isn't the right name for this abstraction, but rather something like `PluginManager`.

Reply:
Sounds okay to me. You can never have too many managers... okay, perhaps you can.

Reply:
I did the rename in a separate commit because then the file renaming is handled better in a diff because the files are similar enough: https://github.com/kubernetes/kubernetes/pull/132096/commits/dec98e8eea5569ad6f43f037032096fe2352d359